### ISSUE TITLE
Improve Training Loop

### ISSUE THREAD

**Issue by zparis99** [leoniekerken/ECoG-foundation-model#18]
The code for the training loop is fairly confusing and messy, and could be improved. Instead of training based on epochs we can limit by the number of training steps we want, making plotting optional, configuring what we want to do at every checkpoint, adding tests, etc.

**Comment by zparis99** [comment]
So after running training there are various issues in the loop working on fixing the following:

1. Currently masks are shared across all examples in a batch. This leads to issues with padding because one example in a batch could be padded (either electrode held out or reached end of file in middle of batch) and causes too much data to be masked out.
2. Currently signal rearranging when calculating losses assumes that the number of patches is divisible by 10 which does not allow for different masking ratios and can also run into issues with padding. This assumption should be relaxed although there's still some work to figure out how data is reconstructed.
3. The masking logic is fairly complicated and probably should be separated from the training loop and moved into the model code. Adding that there.

**Comment by zparis99** [comment]
I have made several improvements to this in both the main branch and by supporting the mae-st code in the mae-st branch. I am more confident in the mae-st code so I would recommend using that going forward.
